codegen/smithy-aws-typescript-codegen/src/main/java/software/amazon/smithy/aws/typescript/codegen/AwsRestXml.java

```diff
@@ -207,7 +208,8 @@ protected void serializeInputPayload(
                     getInputValue(context, Location.PAYLOAD, "input." + memberName, member, target));
 
             // Structure and Union payloads will serialize as XML documents via XmlNode.
-            if (target instanceof StructureShape || target instanceof UnionShape) {
+            if ((target instanceof StructureShape || target instanceof UnionShape)
+                    && member.hasTrait(EventStreamTrait.class)) {
```

Comment:
Should this be inverted to `!member.hasTrait` ?

Reply:
Nice catch!